src/traces/funnel/hover.js

```diff
@@ -0,0 +1,57 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var hoverLabelText = require('../../plots/cartesian/axes').hoverLabelText;
+var opacity = require('../../components/color').opacity;
+var hoverOnBars = require('../bar/hover').hoverOnBars;
+
+module.exports = function hoverPoints(pointData, xval, yval, hovermode) {
+    var point = hoverOnBars(pointData, xval, yval, hovermode);
+    if(!point) return;
+
+    var cd = point.cd;
+    var trace = cd[0].trace;
+    var isHorizontal = (trace.orientation === 'h');
+
+    var vAxis = isHorizontal ? pointData.xa : pointData.ya;
+
+    function formatNumber(a) {
+        return hoverLabelText(vAxis, a);
+    }
+
+    // the closest data point
+    var index = point.index;
+    var di = cd[index];
+
+    var sizeLetter = isHorizontal ? 'x' : 'y';
+
+    point[sizeLetter + 'LabelVal'] = di.s;
+
+    // display ratio to initial value
+    point.extraText = [
+        formatNumber(0.1 * (Math.round(1000 * di.begR))) + '% of initial',
+        formatNumber(0.1 * (Math.round(1000 * di.difR))) + '% of previous',
+        formatNumber(0.1 * (Math.round(1000 * di.sumR))) + '% of total'
+    ].join('<br>');
+    // TODO: Should we use pieHelpers.formatPieValue instead ?
+
+    point.color = getTraceColor(trace, di);
+
+    return [point];
```

Comment:
![Peek 2019-04-29 17-50](https://user-images.githubusercontent.com/6675409/56929338-533b7900-6aa7-11e9-8457-1165f79d2afc.gif)


I think the default hover label looks a little cluttered. Maybe we shouldn't show that position label and/or show 2 of those 3 percentages?

Reply:
I think this is fine, and the user can pare down to what they like. The vast majority of funnels will be single-trace, so it won't seem that bad.

Reply:
> I think this is fine, and the user can pare down to what they like. The vast majority of funnels will be single-trace, so it won't seem that bad.

I would be in favour of a different `hoverinfo` default for traces that appear on subplots with multiple `funnel` traces.

Reply:
ok

Reply:
Closing this one. No need to this!